drivers/sensor/st/lsm6dsv16x/lsm6dsv16x.c

```diff
@@ -1099,6 +1128,45 @@ static int lsm6dsv16x_init(const struct device *dev)
 	return 0;
 }
 
+#if defined(CONFIG_PM_DEVICE)
+static int lsm6dsv16x_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	const struct lsm6dsv16x_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	int ret = 0;
+
+	LOG_DBG("PM action: %d", (int)action);
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		if (lsm6dsv16x_xl_data_rate_set(ctx, cfg->accel_odr) < 0) {
```

Comment:
I think that suspend/resume should restore the value that is currently set in registers, not the one that is in config. This because the value might have been changed by attr_set() after the initial configuration.

Reply:
Ah yes, this is a mistake! I meant to use the values in the data struct here.

Reply:
Yes, exactly